web/src/saved_snippets.ts

```diff
@@ -0,0 +1,51 @@
+import type {Option} from "./dropdown_widget";
+import {$t} from "./i18n";
+import * as util from "./util";
+
+export type SavedSnippet = {
+    id: number;
+    title: string;
+    content: string;
+    date_created: number;
+};
+
+export const ADD_SAVED_SNIPPET_OPTION_ID = -1;
+let my_saved_snippets: SavedSnippet[] = [];
+
+export function set_saved_snippets(saved_snippets: SavedSnippet[]): void {
+    my_saved_snippets = saved_snippets;
+    my_saved_snippets.sort((a, b) => util.strcmp(a.title.toLowerCase(), b.title.toLowerCase()));
+}
+
+export function get_saved_snippets(): SavedSnippet[] {
+    return my_saved_snippets;
+}
+
+export function add_saved_snippet(saved_snippet: SavedSnippet): void {
+    my_saved_snippets.push(saved_snippet);
+    my_saved_snippets.sort((a, b) => util.strcmp(a.title.toLowerCase(), b.title.toLowerCase()));
```

Comment:
Probably this sort belongs in `get_options_for_dropdown_widget`.

While the data sizes are small enough that it likely doesn't matter performance-wise, one should always use the right data structures for what you're doing by default.